src/datasets/datasets-public.v4.controller.ts

```diff
@@ -0,0 +1,245 @@
+import { Controller, Get, Param, Query, HttpStatus } from "@nestjs/common";
+import {
+  ApiExtraModels,
+  ApiOperation,
+  ApiParam,
+  ApiQuery,
+  ApiResponse,
+  ApiTags,
+} from "@nestjs/swagger";
+import { DatasetsService } from "./datasets.service";
+import { DatasetDocument } from "./schemas/dataset.schema";
+import {
+  IDatasetFields,
+  IDatasetFiltersV4,
+} from "./interfaces/dataset-filters.interface";
+import { IFilters } from "src/common/interfaces/common.interface";
+
+import { HistoryClass } from "./schemas/history.schema";
+import { TechniqueClass } from "./schemas/technique.schema";
+import { RelationshipClass } from "./schemas/relationship.schema";
+import { OutputDatasetDto } from "./dto/output-dataset.dto";
+import { CountApiResponse } from "src/common/types";
+import { DatasetLookupKeysEnum } from "./types/dataset-lookup";
+import { IncludeValidationPipe } from "./pipes/include-validation.pipe";
+import { FilterValidationPipe } from "./pipes/filter-validation.pipe";
+import { getSwaggerDatasetFilterContent } from "./types/dataset-filter-content";
+
+@ApiExtraModels(HistoryClass, TechniqueClass, RelationshipClass)
+@ApiTags("datasets public v4")
+@Controller({ path: "datasets/public", version: "4" })
+export class DatasetsPublicV4Controller {
```

Comment:
As we discussed during your demo, naming of v4 may not necessary. Users/Devs can choose image version s to accommodate their needs.
Again, this can be further discussed before merge into master with Max